# file v4l-dvb-saa716x-4.5.2-fix-compile.patch
fix compile with kernel >= 4.5.2
wrt bug 581450
Signed-of-by: Joerg Bornkessel <hd_brummy@gentoo.org> 2016/05/05
--- a/linux/drivers/media/common/saa716x/saa716x_pci.c.old	2016-04-27 20:59:51.000000000 +0200
+++ b/linux/drivers/media/common/saa716x/saa716x_pci.c	2016-04-27 21:19:03.117822874 +0200
@@ -1,6 +1,7 @@
 #include <asm/atomic.h>
 #include <linux/spinlock_types.h>
 #include <asm/io.h>
+#include <asm/processor.h>
 #include <asm/pgtable.h>
 #include <asm/page.h>
 #include <linux/kmod.h>



# file v4l-dvb-saa716x-4.6.0-fix-compile.patch
--- a/linux/drivers/media/common/saa716x/saa716x_pci.c	2017-02-26 18:48:27.196949325 +0100
+++ b/linux/drivers/media/common/saa716x/saa716x_pci.c	2017-02-26 18:50:28.149283794 +0100
@@ -1,9 +1,6 @@
 #include <asm/atomic.h>
 #include <linux/spinlock_types.h>
-#include <asm/io.h>
 #include <asm/processor.h>
-#include <asm/pgtable.h>
-#include <asm/page.h>
 #include <linux/kmod.h>
 #include <linux/vmalloc.h>
 #include <linux/init.h>
@@ -20,11 +17,6 @@
 
 #define DRIVER_NAME				"SAA716x Core"
 
-static irqreturn_t saa716x_msi_handler(int irq, void *dev_id)
-{
-	return IRQ_HANDLED;
-}
-
 static int saa716x_enable_msi(struct saa716x_dev *saa716x)
 {
 	struct pci_dev *pdev = saa716x->pdev;



# file v4l-dvb-saa716x-4.9-fix-warnings.patch
--- a/linux/drivers/media/common/saa716x/saa716x_rom.c	2015-12-29 19:40:55.000000000 +0100
+++ a/linux/drivers/media/common/saa716x/saa716x_rom.c	2017-07-11 00:51:19.149941914 +0200
@@ -113,7 +113,7 @@
 	memcpy(rom_header, &buf[*offset], sizeof (struct saa716x_romhdr));
 	if (rom_header->header_size != sizeof (struct saa716x_romhdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%zd bytes, Expected=%d",
 			sizeof (struct saa716x_romhdr),
 			rom_header->header_size);
 
@@ -237,7 +237,7 @@
 	saa716x_descriptor_dbg(saa716x, buf, offset, header.size, header.ext_data);
 	if (header.size != sizeof (struct saa716x_decoder_hdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%zd",
 			header.size,
 			sizeof (struct saa716x_decoder_hdr));
 
@@ -268,7 +268,7 @@
 	saa716x_descriptor_dbg(saa716x, buf, offset, header.size, header.ext_data);
 	if (header.size != sizeof (struct saa716x_gpio_hdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%zd",
 			header.size,
 			sizeof (struct saa716x_gpio_hdr));
 
@@ -305,7 +305,7 @@
 	saa716x_descriptor_dbg(saa716x, buf, offset, header.size, header.ext_data);
 	if (header.size != sizeof (struct saa716x_video_decoder_hdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%zd",
 			header.size,
 			sizeof (struct saa716x_video_decoder_hdr));
 
@@ -386,7 +386,7 @@
 	saa716x_descriptor_dbg(saa716x, buf, offset, header.size, header.ext_data);
 	if (header.size != sizeof (struct saa716x_audio_decoder_hdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%zd",
 			header.size,
 			sizeof (struct saa716x_audio_decoder_hdr));
 
@@ -417,7 +417,7 @@
 	saa716x_descriptor_dbg(saa716x, buf, offset, header.size, header.ext_data);
 	if (header.size != sizeof (struct saa716x_evsrc_hdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%zd",
 			header.size,
 			sizeof (struct saa716x_evsrc_hdr));
 
@@ -448,7 +448,7 @@
 	saa716x_descriptor_dbg(saa716x, buf, offset, header.size, header.ext_data);
 	if (header.size != sizeof (struct saa716x_xbar_hdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%zd",
 			header.size,
 			sizeof (struct saa716x_xbar_hdr));
 
@@ -486,7 +486,7 @@
 	saa716x_descriptor_dbg(saa716x, buf, offset, header.size, header.ext_data);
 	if (header.size != sizeof (struct saa716x_tuner_hdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%zd",
 			header.size,
 			sizeof (struct saa716x_tuner_hdr));
 
@@ -516,7 +516,7 @@
 	saa716x_descriptor_dbg(saa716x, buf, offset, header.size, header.ext_data);
 	if (header.size != sizeof (struct saa716x_pll_hdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%zd",
 			header.size,
 			sizeof (struct saa716x_pll_hdr));
 
@@ -546,7 +546,7 @@
 	saa716x_descriptor_dbg(saa716x, buf, offset, header.size, header.ext_data);
 	if (header.size != sizeof (struct saa716x_channel_decoder_hdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%zd",
 			header.size,
 			sizeof (struct saa716x_channel_decoder_hdr));
 
@@ -576,7 +576,7 @@
 	saa716x_descriptor_dbg(saa716x, buf, offset, header.size, header.ext_data);
 	if (header.size != sizeof (struct saa716x_encoder_hdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%zd",
 			header.size,
 			sizeof (struct saa716x_encoder_hdr));
 
@@ -606,7 +606,7 @@
 	saa716x_descriptor_dbg(saa716x, buf, offset, header.size, header.ext_data);
 	if (header.size != sizeof (struct saa716x_ir_hdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%zd",
 			header.size,
 			sizeof (struct saa716x_ir_hdr));
 
@@ -637,7 +637,7 @@
 	saa716x_descriptor_dbg(saa716x, buf, offset, header.size, header.ext_data);
 	if (header.size != sizeof (struct saa716x_eeprom_hdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%zd",
 			header.size,
 			sizeof (struct saa716x_eeprom_hdr));
 
@@ -668,7 +668,7 @@
 	saa716x_descriptor_dbg(saa716x, buf, offset, header.size, header.ext_data);
 	if (header.size != sizeof (struct saa716x_filter_hdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%zd",
 			header.size,
 			sizeof (struct saa716x_filter_hdr));
 
@@ -699,7 +699,7 @@
 	saa716x_descriptor_dbg(saa716x, buf, offset, header.size, header.ext_data);
 	if (header.size != sizeof (struct saa716x_streamdev_hdr)) {
 		dprintk(SAA716x_ERROR, 1,
-			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%d",
+			"ERROR: Header size mismatch! Read size=%d bytes, Expected=%zd",
 			header.size,
 			sizeof (struct saa716x_streamdev_hdr));
 
@@ -789,7 +789,7 @@
 
 	memcpy(device, &buf[*offset], sizeof (struct saa716x_devinfo));
 	if (device->struct_size != sizeof (struct saa716x_devinfo)) {
-		dprintk(SAA716x_ERROR, 1, "ERROR: Device size mismatch! Read=%d bytes, expected=%d bytes",
+		dprintk(SAA716x_ERROR, 1, "ERROR: Device size mismatch! Read=%d bytes, expected=%zd bytes",
 		device->struct_size,
 		sizeof (struct saa716x_devinfo));
 
